lib/Db/WopiMapper.php

```diff
@@ -146,18 +154,14 @@ public function getWopiForToken($token) {
 			'app' => 'richdocuments'
 		]);
 		if ($row === false) {
-			throw new DoesNotExistException('Could not find token.');
+			throw new UnknownTokenException('Could not find token.');
 		}
 
 		/** @var Wopi $wopi */
 		$wopi = Wopi::fromRow($row);
 
-		//TODO: validate.
-		if ($wopi->getExpiry() > $this->timeFactory->getTime()){
-			// Expired token!
-			//http_response_code(404);
-			//$wopi->deleteBy('id', $row['id']);
-			//return false;
+		if ($wopi->getExpiry() < $this->timeFactory->getTime()){
```

Comment:
Shall we clean up the token right away here as well assuming that Collabora properly indicates the token invalidation and does not retry other operations with the same token afterwards? Otherwise I'd also be fine to rely on the background job for the cleanup.

Reply:
Collabora will continue using an expired token. 

For example if the user clicks the "save" button collabora will try to save it with the expired token, then if it receives a 401 it will show this messagee
![test-odt - Files - Nextcloud 4-19-2022 5-20-06 PM](https://user-images.githubusercontent.com/3178979/164038545-f5ea6707-fd7d-451b-a48a-343e0d3e9d6b.png)

But the user can continue to edit the doocument and can even download the document.

Reply:
All right, then lets stick with that as it is and probably suggest to improve that on the upstream side at https://github.com/CollaboraOnline/online